src/boxed.rs

```diff
@@ -180,15 +179,14 @@ macro_rules! glib_boxed_wrapper {
                 res
             }
 
-            unsafe fn from_glib_full_num_as_vec(mut ptr: *mut *mut $ffi_name, num: usize) -> Vec<Self> {
+            unsafe fn from_glib_full_num_as_vec(ptr: *mut *mut $ffi_name, num: usize) -> Vec<Self> {
                 if num == 0 || ptr.is_null() {
                     return Vec::new();
                 }
 
                 let mut res = Vec::with_capacity(num);
-                for _ in 0..num {
-                    res.push($crate::translate::from_glib_full(ptr::read(ptr)));
-                    ptr = ptr.offset(1);
+                for i in 0..num {
+                    res.push($crate::translate::from_glib_full(ptr::read(ptr.offset(i as isize))));
                 }
                 glib_ffi::g_free(ptr as *mut _);
```

Comment:
We're copying everything here anyway (which we don't really need if we had more infrastructure) :) And it shouldn't be slower really: first of all, a sufficiently smart compiler should otherwise be able to optimize it that way, and more importantly: the difference is between +1, and +1 for the iterator and +n for the pointer. That's one extra instruction (register +1) and one changed instruction (pointer addition by register instead of immediate value). That hardly matters.

I didn't check the assembly output though, it doesn't seem worth thinking about this here.